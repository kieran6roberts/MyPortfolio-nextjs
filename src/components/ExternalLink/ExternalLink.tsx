import { BsBoxArrowUpRight } from "react-icons/bs";

interface EXTERNAL_LINK {
    children: React.ReactNode;
    styling?: string | undefined;
    link: string;
};

export default function ExternalLink({ link, children, styling }: EXTERNAL_LINK): React.ReactElement {
    return (
        <a href={link}
        rel="noopener"
        target="_blank"
        className={`group ${styling ?? null} w-44 2xl:w-80 flex align-items whitespace-nowrap text-xxs text-dark font-bold uppercase py-2 px-3 2xl:py-6 2xl:px-6 border-2 border-pri cursor-pointer transition duration-50 ease-in hover:bg-pri focus:outline-none focus:ring-4 focus:ring-yellow-400 active:bg-sec`} >
            <span className="block pr-4 group-hover:text-light ">
                {children}
            </span>
            <BsBoxArrowUpRight className="text-xs transition ease-in text-pri duration-50 group-hover:text-white"/>
        </a>
    )
};