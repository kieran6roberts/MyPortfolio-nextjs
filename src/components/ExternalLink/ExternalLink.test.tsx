import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import ExternalLink from "@/components/ExternalLink/ExternalLink";

let documentBody: RenderResult;

describe("<ExternalLink />", () => {
    test("renders", () => {
        documentBody = render(<ExternalLink styling="justify-center" link="/">child</ExternalLink>);
        expect(documentBody.getByRole("link")).toBeInTheDocument();
        expect(documentBody.getByRole("link")).toHaveClass("justify-center");
        expect(documentBody.getByText("child")).toBeInTheDocument();
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<ExternalLink styling="justify-center" link="/">child</ExternalLink>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});