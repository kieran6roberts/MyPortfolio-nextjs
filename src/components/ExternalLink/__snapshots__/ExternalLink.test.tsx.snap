// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink /> render snapshot 1`] = `
<a
  className="group justify-center w-44 2xl:w-80 flex align-items whitespace-nowrap text-xxs text-dark font-bold uppercase py-2 px-3 2xl:py-6 2xl:px-6 border-2 border-pri cursor-pointer transition duration-50 ease-in hover:bg-pri focus:outline-none focus:ring-4 focus:ring-yellow-400 active:bg-sec"
  href="/"
  rel="noopener"
  target="_blank"
>
  <span
    className="block pr-4 group-hover:text-light "
  >
    child
  </span>
  <svg
    className="text-xs transition ease-in text-pri duration-50 group-hover:text-white"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M1.5 13A1.5 1.5 0 003 14.5h8a1.5 1.5 0 001.5-1.5V9a.5.5 0 00-1 0v4a.5.5 0 01-.5.5H3a.5.5 0 01-.5-.5V5a.5.5 0 01.5-.5h4a.5.5 0 000-1H3A1.5 1.5 0 001.5 5v8zm7-11a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V2.5H9a.5.5 0 01-.5-.5z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M14.354 1.646a.5.5 0 010 .708l-8 8a.5.5 0 01-.708-.708l8-8a.5.5 0 01.708 0z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;
